src/array/methods.jl

```diff
@@ -547,6 +547,100 @@ $message on dimension $D.
 To fix for `AbstractDimArray`, pass new lookup values as `cat(As...; dims=$D(newlookupvals))` keyword or `dims=$D()` for empty `NoLookup`.
 """
 
+function Base._typed_stack(::Colon, ::Type{T}, ::Type{S}, A, Aax=_iterator_axes(A)) where {T,S<:AbstractDimArray}
```

Comment:
How stable do you think these methods are... could we add a method to use `Base.stack` instead? What do we can from touching these internals?

I know we use internals elsewhere, but we should stop:
https://github.com/rafaqz/DimensionalData.jl/issues/522